### PR TITLE
Add explainer on entities and domains

### PR DESCRIPTION
**Pull request by c0ffeeca7**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive guide on entities and domains within the Home Assistant ecosystem, enhancing user understanding of device management and control.
	- Added a navigation link to the "Entities and domains" section in the documentation for improved accessibility.
	- Enhanced the state objects documentation with a new "related" section for better contextual navigation.

### CODE REVIEW

File: source/_docs/configuration/entities_domains.markdown [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+title: "Entities and domains"
+description: "Describes what entities and domains are in Home Assistant."
+related:
+  - docs: /docs/configuration/state_object/
+    title: State object, entity state
+---
+
+Your {% term devices %} are represented in Home Assistant as entities. Entities are the basic building blocks to hold data in Home Assistant. An entity represents a {% term sensor %}, actor, or function in Home Assistant. Entities are used to monitor physical properties or to control other entities. An term entity is usually part of a {% term device %} or a {% term service %}. Entities have [states](/docs/configuration/state_object/) and [attributes](#entity-attributes).
```
Comment (c0ffeeca7) [inline source/_docs/configuration/entities_domains.markdown]:
```suggestion
Your {% term devices %} are represented in Home Assistant as entities. Entities are the basic building blocks to hold data in Home Assistant. An entity represents a {% term sensor %}, actor, or function in Home Assistant. Entities are used to monitor physical properties or to control other entities. An entity is usually part of a {% term device %} or a {% term service %}. Entities have [states](/docs/configuration/state_object/) and [attributes](#entity-attributes).
```

File: source/_docs/configuration/entities_domains.markdown [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+<p class='img'>
+  <img src='/images/integrations/light/state_light.png' alt='Screenshot showing three lights with different states and attributes'>
+  Example showing three lights with differents states and different attributes.
+</p>
+
+For more information on entity attributes, refer to the [attributes](/docs/configuration/state_object/#attributes) section on the state objects page.
+
+## Domains
+
+Each integration in Home Assistant has a unique identifier: a domain. All entities and actions available in Home Assistant are provided by integrations and thus belong to such a domain. The first part of the entity or action, before the . shows the domain they belong to. For example, `light.bed_light` is an entity in the light domain. `bed_light` is the name of the entity.
```
Comment (klaasnicolaas) [inline source/_docs/configuration/entities_domains.markdown]: Are we talking about integration domain or entity domain here?

Comment (c0ffeeca7) [reply source/_docs/configuration/entities_domains.markdown]: excellent question... let me take another stab at this....

File: source/_docs/configuration/entities_domains.markdown [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+<p class='img'>
+  <img src='/images/integrations/light/state_light.png' alt='Screenshot showing three lights with different states and attributes'>
+  Example showing three lights with differents states and different attributes.
+</p>
+
+For more information on entity attributes, refer to the [attributes](/docs/configuration/state_object/#attributes) section on the state objects page.
+
+## Domains
+
+Each integration in Home Assistant has a unique identifier: a domain. All entities and actions available in Home Assistant are provided by integrations and thus belong to such a domain. The first part of the entity or action, before the `.` shows the domain they belong to. For example, `light.bed_light` is an entity in the light domain. `bed_light` is the name of the entity.
```
Comment (marcelveldt) [inline source/_docs/configuration/entities_domains.markdown]:
```suggestion
Each integration in Home Assistant has a unique identifier: a domain. All entities and actions available in Home Assistant are provided by integrations and thus belong to such a domain. The first part of the entity or action, before the `.` shows the domain they belong to. For example, `light.bed_light` is an entity in the light domain. `bed_light` is the ID of the entity.
```

File: source/_docs/configuration/entities_domains.markdown [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+All your entities are listed in the entities table, under {% my entities title="**Settings** > **Devices & services** > **Entities**" %}.
+
+<p class='img'><img src='/images/getting-started/entities.png' style='border: 0;box-shadow: none;' alt="Screenshot showing the Entities table">Screenshot of the Entities table. Each line represents an entity.</p>
+
+## Entity attributes
+
+Many entities have attributes. There are a few attributes that are used by Home Assistant for representing the entity in a specific way. Each integration will also have its own attributes to represent extra state data about the entity. For example, the light integration has attributes for the current brightness and color of the light. When an attribute is not available, Home Assistant will not write it to the state. The list of available attributes depends on the {% term device %}.
```
Comment (MartinHjelmare) [inline source/_docs/configuration/entities_domains.markdown]: Maybe give an example of these kinds of attributes:

> There are a few attributes that are used by Home Assistant for representing the entity in a specific way.

Comment (MartinHjelmare) [reply source/_docs/configuration/entities_domains.markdown]: I also think this sentence is hard to understand. I think we should rewrite it, but I'd like to get an example first and maybe that will clarify how to explain it.

Comment (c0ffeeca7) [reply source/_docs/configuration/entities_domains.markdown]: Good question. I'm not actually sure what this is referring to...I copied it from here: <https://www.home-assistant.io/docs/configuration/state_object/#attributes>. Maybe it's best to remove that sentence, on both pages?

Comment (MartinHjelmare) [reply source/_docs/configuration/entities_domains.markdown]: I think it's referring to the common state attributes that are available on all entities, eg `friendly_name` and `icon`.

Comment (c0ffeeca7) [reply source/_docs/configuration/entities_domains.markdown]: Oh, I see. Thank you!

**Review by MartinHjelmare — APPROVED** [review]
Great!

File: source/_docs/configuration/entities_domains.markdown [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+All your entities are listed in the entities table, under {% my entities title="**Settings** > **Devices & services** > **Entities**" %}.
+
+<p class='img'><img src='/images/getting-started/entities.png' style='border: 0;box-shadow: none;' alt="Screenshot showing the Entities table">Screenshot of the Entities table. Each line represents an entity.</p>
+
+## Entity attributes
+
+Many entities have attributes. There are a few attributes that are available on all entities, such as `friendly_name` and `icon`. In addition to those, each integration has its own attributes to represent extra state data about the entity. For example, the light integration has attributes for the current brightness and color of the light. When an attribute is not available, Home Assistant will not write it to the state. The list of available attributes depends on the {% term device %}.
```
Comment (MartinHjelmare) [inline source/_docs/configuration/entities_domains.markdown:17]: I wonder if we need to specify the common attributes better. The common attributes can be set on all entities, but they could still be missing if they are not set.

Comment (c0ffeeca7) [reply source/_docs/configuration/entities_domains.markdown:17]: Thank you for the hint! 
https://github.com/home-assistant/home-assistant.io/pull/34975

**Pull request merged**